run.py

```diff
@@ -106,6 +108,7 @@ def main():
     #  It’s a no-op if the 'gpu_index' argument is a negative integer or None.
     with torch.cuda.device(args.gpu_index):
         enn = ENN(model=student_model, optimizer=optimizer, args=args)
+        enn.train_gating_networks(train_loader, valid_loader)
```

Comment:
No flag, and calling this before enn.train() ?

Reply:
I will add the flag in the future. In general code of the repo requires cleaning up. The main reason for this PR is just to showcase the differences/code for the training of gating networks. I won't merge it before clean up.